code/API_definitions/device-status.yaml

```diff
@@ -205,6 +466,244 @@ components:
+        - type: object
+          properties:
+            eventDetail:
+              $ref: '#/components/schemas/RoamingStatus'
+
+    EventRoamingOn:
+      description:  event structure for roaming on and roaming off change
+      allOf:
+        - $ref: '#/components/schemas/Event'
+        - type: object
+          properties:
+            eventType:
+              type: string
+              enum:
+                - ROAMING_ON
+            eventDetail:
+              $ref: '#/components/schemas/RoamingChange'
+
+    EventRoamingOff:
+      description:  event structure for roaming on and roaming off change
+      allOf:
+        - $ref: '#/components/schemas/Event'
+        - type: object
+          properties:
+            eventType:
+              type: string
+              enum:
+                - ROAMING_OFF
+            eventDetail:
+              $ref: '#/components/schemas/RoamingChange'
+
+    EventChangeCountry:
+      description:  event structure for roaming change country
+      allOf:
+        - $ref: '#/components/schemas/Event'
+        - type: object
+          properties:
+            eventType:
+              type: string
+              enum:
+                - ROAMING_CHANGE_COUNTRY
+            eventDetail:
+              $ref: '#/components/schemas/RoamingChangeCountry'
+
+    EventSubscriptionEnds:
+      description:  event structure for event subscription ends
+      allOf:
+        - $ref: '#/components/schemas/Event'
+        - type: object
+          properties:
+            eventDetail:
+              $ref: '#/components/schemas/SubscriptionEnds'
+
+    RoamingStatus:
+      description: Event detail structure for ROAMING_STATUS event
+      type : object
+      required:
+        - device
+        - roaming
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        roaming:
+          type: boolean
+          description:  Roaming status. True, if it is roaming.
+        countryCode:
+          $ref: '#/components/schemas/CountryCode'
+        countryName:
+          $ref: '#/components/schemas/CountryName'
+
+    RoamingChange:
+      description: Event detail structure for ROAMING_ON & ROAMING_OFF event
+      type : object
+      required:
+        - device
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+
+    RoamingChangeCountry:
+      description: Event detail structure for ROAMING_CHANGE_COUNTRY event
+      type : object
+      required:
+        - device
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        countryCode:
+          $ref: '#/components/schemas/CountryCode'
+        countryName:
+          $ref: '#/components/schemas/CountryName'
+
+    SubscriptionEnds:
+      description: Event detail structure for SUBSCRIPTION_ENDS event
+      type : object
+      required:
+        - device
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+
```

Comment:
I think this eventDetail should contain the property terminationReason

Reply:
This would be an additional property in the `SubscriptionEnds` schema.

But what reasons are there for the subscription ending other than it has expired (i.e. the `subscriptionExpireTime` has been reached)? I can see that there may be other reasons in theory, but I don't see that they will be encountered in practice. Why would the API provider choose to terminate the subscription early?

In any case, the API client can infer "other reasons" for subscription termination if `subscriptionExpireTime` has not been reached.

So I don't have a problem with adding a `terminationReason` property, but what would be the reasons other than `SUBSCRIPTION_EXPIRED`?

Reply:
OK, fair point that this is a requirement of the CAMARA API Design Guidelines.

The example used in the design guidelines suggests that `terminationReason` is just a freeform string. So can we just add the property with no pre-defined values? Or should it be an enum, maybe {`SUBSCRIPTION_EXPIRED`, `NETWORK_TERMINATED`}?

Reply:
Yes, it could be both... I tend to use enums, it's more uniform through all the APIs and can be evaluated programmatically

Reply:
OK, I added the terminationReason as an enum